internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -195,11 +216,22 @@ func (c *enrollCmd) Execute(ctx context.Context, streams *cli.IOStreams) error {
 		span.End()
 	}()
 
-	hasRoot, err := utils.HasRoot()
+	hasRoot, err := c.hasRoot()
 	if err != nil {
 		return fmt.Errorf("checking if running with root/Administrator privileges: %w", err)
 	}
 
+	c.log.Debug("verifying communication with running Elastic Agent daemon")
+	// attempt to get the agent state, if there is an error, proceed without
+	// handling the error and provide the fleetServerBootstrap with the error
+	as, asErr := c.getAgentStateWB(ctx)
+	if asErr == nil {
+		// skip this check if enroll is run by the install command
+		if hasRoot && as.Info.Unprivileged && !c.options.FromInstall {
+			return unprivilegedAgentRootUserError
+		}
+	}
```

Comment:
I did not see the original filled issue for this and I would have made a comment there. I know this is coming late, but I believe this is the wrong approach.

If you call `enroll` as root, you can easily tell if Elastic Agent is installed without communicating with the daemon. You can read the user/group from the executing binary from the filesystem, because you are root. 

You can then just `ReExec` the enroll command directly using the `user` you retrieved from reading from the disk. You need to read `user` from the filesystem as @michalpristas is adding the ability to use a custom user, so you cannot rely on the idea that the user is always the same.

This change also will allow a root user to call enroll and have the enroll work correctly as the unprivileged user.

Reply:
Yes it would, and I am now thinking that the current implementation is tricky to get working with delayed enrolls. I started looking into your suggestion and will be replacing this PR.